lambdaclass/curse_of_mirra
# [GH-1000] Select character functionality

Closes #1000 

Works in conjunction with [this backend PR](https://github.com/lambdaclass/game_backend/pull/61).

Waiting on #1237 

## Motivation

Users must be able to select the character they want to play with.

## How has this been tested?

#### To test it is needed to enable one or more characters, since in this PR Muflus is the only one available. This is done by selecting the scripteable object of the character you want to enable and check the "Enabled" option.
#### It's also needed to modify the Utils script to point towards the localhost server before getting to the main screen (where you can choose the server manually). This is done by changing the `MakeHTTPUrl` function of the class to:

```
    public static string MakeHTTPUrl(string path)
    {
        return "http://" + "localhost" + ":4000" + path;
    }
```

There is a simplify user creation flow explanation:
![UserCreation drawio](https://github.com/lambdaclass/curse_of_mirra/assets/62818001/c1fe841c-d91f-460a-8e4e-bd89a43a949e)

There is a simplify character Selection flow explanation:
<img width="882" alt="Captura de pantalla 2023-12-13 a la(s) 13 31 26" src="https://github.com/lambdaclass/curse_of_mirra/assets/62818001/10589c8f-9582-46ee-b0d9-242ab11c1ec0">


- To test user creation on title screen:
  - You must not have an user in the database with your `device_client_id`.
  - Start the game and after you click the ENTER button in title screen scene a loading spinner should appear, then you will be sent to main screen and a user with your `device_client_id` should have been created in the database.
- To test character selection:
  - Go to character list scene, choose one and select it in the character info scene.
  - You will be taken to the main scene and the character model to appear in the middle of the screen should be the one you chose.
  - Also the `selected_character` column in your database user should be the character chosen.
  - When you play a game the character you are assigned should be the one you have selected.

## Test additions / changes

1. Enable H4ck adding "H4ck" to the `availableCharacterNames` list in `CharactersManager`
2. In the scriptableObject of H4ck add the Muflus prefab to Prefab field and Muflus UImodel to the UImodel field (This is because we dont have th h4ck model anymore)
<img width="311" alt="Captura de pantalla 2023-12-21 a la(s) 12 25 00" src="https://github.com/lambdaclass/curse_of_mirra/assets/62818001/bff0cb35-4ba6-41af-9a1e-92aae8a2a1fb">


3. Run the backend and the client
4. Go to characterList select Muflus Or H4ck you should see the name of the selected in the LobbyConnection GO in the field `selectedCharacterName`
5. Play a game you should have the selected
6.  Kill the client. Start again you should have the last selected character as you starter character
7. Repeat


## Checklist
